### ISSUE TITLE
Y axis elevation doesn't add path points

### ISSUE THREAD

**Issue by rustwork** [MilosLukic/Godot-Navigation-Lite#5]
If there is no hard obstacle to go around in the X/Z axis, then the navigation always draws a straight line to the target, even when there is elevation in the Y axis that the path should climb over. This causes the character to move underground through hills etc. 

Here's a screenshot from your example project, where I changed the navmesh to have **Tile_Size=8** and **Agent_Max_Climb=3**
You can see that the line goes straight through the hill created by the spheres.

![navbug](https://user-images.githubusercontent.com/2095704/95058115-ebca2580-06ab-11eb-89d7-0ee93674a117.JPG)

**Comment by MilosLukic** [comment]
Yes, I noticed this behaviour, but I didnt have time to find out the reasons for it (It could be the simplification logic for the path, not sure). I didnt pay much attention because with physics set up, the character would move up the hill. I know it's a little bit hacky, and I'm very much open to accepting PR's that enable more accurate paths.

**Comment by Pyredrid** [comment]
Is there any chance of this bug getting fixed?  It's affecting my project as well, and so far all my attempts to fix it have failed worse.  The lack of comments in the codebase doesn't help with understanding what's going on...

**Comment by Pyredrid** [comment]
Submitted a Pull Request that should fix this bug.  If anyone has a Windows environment and can double check it builds correctly there then it should be good to go.
